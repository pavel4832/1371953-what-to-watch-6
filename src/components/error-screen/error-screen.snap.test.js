import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import ErrorScreen from './error-screen';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import MOCK_STORE from '../../mock/mock-store';

const mockStore = configureStore({});

it(`Should ErrorScreen render correctly`, () => {
  const history = createMemoryHistory();
  const store = mockStore(MOCK_STORE);

  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <ErrorScreen />
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
