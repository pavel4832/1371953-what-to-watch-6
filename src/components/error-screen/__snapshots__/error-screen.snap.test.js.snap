// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ErrorScreen render correctly 1`] = `
<div>
  <h1
    class="visually-hidden"
  >
    WTW
  </h1>
  <header
    class="page-header movie-card__head "
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    
    <h1
      class="page-title user-page__title"
      style="display: none;"
    />
    <div
      class="user-block"
    >
      <div
        class="user-block__avatar"
      >
        <a
          href="/mylist"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </a>
      </div>
      <a
        class="user-block__link"
        href="/"
      >
        Log out
      </a>
    </div>
  </header>
  <section
    class="page-header"
    style="flex-direction: column;"
  >
    <h1>
      404. Page not found
    </h1>
    <a
      href="/"
    >
      Вернуться на главную
    </a>
  </section>
  <footer
    class="page-footer"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link logo__link--light"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      class="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
